src/store/reducers/hackathonReducer.js

```diff
@@ -5,7 +5,7 @@ const initialState = {
   assets: [],
   hackDetails: null,
   assignedHacks: [],
-  submission: null
+  submission: null,
 };
 
 const reducer = (state = initialState, action) => {
```

Comment:
you seem to only update one part of this reducer in each action, its a sign that this should probably sliced up into smaller reducers and use `combineReducers` to build up your store reducer.

Reply:
At the moment I feel it's okay to cram this into the hackathon reducer for submission details, however, Joseph is working on the upload functionality for the app and that has yet to be merged, will have a discussion with him to maybe have a reducer for **hack detail** which include submission along with some of the upload state details.

potential new setup:

```javascript
const rootReducer = combineReducers({
  auth: persistReducer(persistConfig, authReducer),
  user: userReducer,
  hack: hackReducer,
  submission: submissionReducer // to include submission details along with any of the upload info
});

```

Reply:
you can nest reducers (if you want... no right or wrong way, your state is yours to organise however works best):

```
const rootReducer = combineReducers({
  auth: persistReducer(persistConfig, authReducer),
  user: userReducer,
  hack: combineReducers({
    details: hackReducer,
    submission: submissionReducer // to include submission details along with any of the upload info
  })
});
```

The point i was trying to make was that you can slice up your reducers smaller and combine them together to make maintenance easier and improve clarity of the code.

You can also do everything with a single reducer... i just wouldn't :P

Reply:
Did not know you can do that, thank you so much!